src/bridges/angle/AngleBridge.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {IStableMaster} from "../../interfaces/angle/IStableMaster.sol";
+
+/**
+ * @title Angle Protocol bridge contract.
+ * @author Angle Protocol Team
+ * @notice Allows a user to deposit/withdraw his tokens to/from Angle Protocol
+ * @dev Enter Angle Protocol as an SLP by depositing authorized tokens
+ */
+contract AngleBridge is BridgeBase {
+    using SafeERC20 for IERC20;
+
+    IStableMaster public constant STABLE_MASTER = IStableMaster(0x5adDc89785D75C86aB939E9e15bfBBb7Fc086A87);
+
+    // collateralAddress => poolManagerAddress
+    mapping(address => address) internal poolManagers;
```

Comment:
Another gas optimization idea which came to my mind is replacing this mapping with `getPoolManager(...)` function which would simply contain few if statements with hardcoded addresses. That way we would avoid doing the SLOAD since those values would simply get copied from bytecode. But I leave that up to you if you want to do it. I did the same optimization in the [Uniswap bridge](https://github.com/AztecProtocol/aztec-connect-bridges/blob/80a8b3dbf1b2c5e302fc2613ec7588108269da34/src/bridges/uniswap/UniswapBridge.sol#L549) and it significantly decreased the costs.

Reply:
Running the tests without this optimization resulted in max consumption of 138562 gas. Let's see how big a difference this will make.
<img width="903" alt="image" src="https://user-images.githubusercontent.com/13470840/183621330-2f31e25c-a41b-4d00-bb1e-d136f011c642.png">